### PR TITLE
Store: dont rely on slice labels continued

### PR DESCRIPTION
**Pull request by MichaHoffmann**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

Converted some other slice label usages to be stringlabel compatible. I fear that performance will take a hit because we are not very economical with our labels.Builders here, but i'm happy if tests pass first of all.

## Verification

Previous Tests still pass hopefully

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -118,20 +118,19 @@ require (
 require (
 	github.com/mitchellh/go-ps v1.0.0
 	github.com/onsi/gomega v1.27.10
-	github.com/prometheus-community/prom-label-proxy v0.7.0
+	github.com/prometheus-community/prom-label-proxy v0.8.1-0.20240127162815-c1195f9aabc0
```
Comment (squat) [inline go.mod:121]: 😎😎😎

Comment (MichaHoffmann) [reply go.mod:121]: Thanks for merging! :bow:

**Review by yeya24 — APPROVED** [review]
Thanks!

File: pkg/store/labelpb/label.go [hunk 2]
```diff
@@ -291,40 +291,14 @@ func (m *ZLabel) Compare(other ZLabel) int {
...
-					b.Add(l.Name, l.Value)
-					break
-				} else if d > 0 {
-					b.Add(e.Name, e.Value)
-					extend = extend[1:]
-				}
-			}
-		}
+	b := labels.NewBuilder(lset)
+	extend.Range(func(l labels.Label) {
```
Comment (fpetkovski) [inline pkg/store/labelpb/label.go:299]: I think this introduces a double for loop, but it might not be as bad in real world cases.

**Review by fpetkovski — APPROVED** [review]

File: pkg/store/labelpb/label.go (hunk 2 above)
Comment (MichaHoffmann) [reply pkg/store/labelpb/label.go:299]: It benchmarks 3x worse for what we have, a solution is to pass in extend still as a slice but that would need way more refactoring.

Comment (fpetkovski) [reply pkg/store/labelpb/label.go:299]: I think this is fine for now, we can optimize if it shows in a profile.

**Pull request merged**